<launch>
  <arg name="file_name" />
  <arg name="lanelet2_map_topic" default="vector_map"/>
  <arg name="lanelet2_map_marker_topic" default="vector_map_marker"/>
  <arg name="center_line_resolution" default="5.0"/>

  <node pkg="map_loader" exec="lanelet2_map_loader" name="lanelet2_map_loader" args="$(var file_name)">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>
    <remap from="output/lanelet2_map" to="$(var lanelet2_map_topic)" />
    <param name="center_line_resolution" value="$(var center_line_resolution)" />
  </node>
  <node pkg="map_loader" exec="lanelet2_map_visualization" name="lanelet2_map_visualization">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>
    <remap from="input/lanelet2_map" to="$(var lanelet2_map_topic)" />
    <remap from="output/lanelet2_map_marker" to="$(var lanelet2_map_marker_topic)" />
  </node>

</launch>
